File: ios/BatteryHolder/Features/BoardSetup/BoardSetupView.swift
import SwiftUI

/// Pick the board you're using and the transport you'll talk to it over.
struct BoardSetupView: View {
    @EnvironmentObject var appState: AppState

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: Theme.spacing.lg) {
                SectionHeader(title: "Choose your board",
                              subtitle: "Pick the module you wired the battery to.")

                ForEach(appState.boards) { board in
                    boardCard(board)
                }

                if let board = appState.selectedBoard {
                    transportSection(for: board)
                    NavigationLink {
                        PinConfigView()
                    } label: {
                        Text("Configure pins")
                    }
                    .buttonStyle(PrimaryButtonStyle())
                    .padding(.top, Theme.spacing.sm)
                }
            }
            .padding(Theme.spacing.lg)
        }
        .background(Theme.color.background)
        .navigationTitle("Setup")
    }

    private func boardCard(_ board: Board) -> some View {
        let isSelected = appState.selectedBoard?.id == board.id
        return Button {
            appState.selectBoard(board)
        } label: {
            Card {
                HStack(alignment: .top, spacing: Theme.spacing.md) {
                    VStack(alignment: .leading, spacing: Theme.spacing.xs) {
                        Text(board.name).font(Theme.font.headline)
                            .foregroundStyle(Theme.color.textPrimary)
                        Text(board.chip.displayName).font(Theme.font.caption)
                            .foregroundStyle(Theme.color.brand)
                        Text(board.summary).font(Theme.font.footnote)
                            .foregroundStyle(Theme.color.textSecondary)
                        HStack(spacing: Theme.spacing.xs) {
                            ForEach(board.supportedTransports) { TransportBadge(transport: $0) }
                        }
                        .padding(.top, Theme.spacing.xxs)
                    }
                    Spacer()
                    Image(systemName: isSelected ? "checkmark.circle.fill" : "circle")
                        .foregroundStyle(isSelected ? Theme.color.brand : Theme.color.border)
                        .font(.title2)
                }
            }
        }
        .buttonStyle(.plain)
    }

    private func transportSection(for board: Board) -> some View {
        VStack(alignment: .leading, spacing: Theme.spacing.sm) {
            SectionHeader(title: "Transport",
                          subtitle: "How the app talks to this board.")
            Picker("Transport", selection: $appState.activeTransport) {
                ForEach(board.supportedTransports) { t in
                    Text(t.displayName).tag(t)
                }
            }
            .pickerStyle(.segmented)
        }
    }
}

#if DEBUG
struct BoardSetupView_Previews: PreviewProvider {
    static var previews: some View {
        Group {
            NavigationStack { BoardSetupView() }
                .environmentObject(AppState.preview)
                .previewDisplayName("Board selected")

            NavigationStack { BoardSetupView() }
                .environmentObject(AppState.previewEmpty)
                .previewDisplayName("Nothing selected")

            NavigationStack { BoardSetupView() }
                .environmentObject(AppState.preview)
                .preferredColorScheme(.dark)
                .previewDisplayName("Dark")
        }
        .tint(Theme.color.brand)
    }
}
#endif

File: android_app/lib/features/board_setup/board_setup_view.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../../app/app_state.dart';
import '../../design_system/components.dart';
import '../../design_system/theme.dart';
import '../../models/board.dart';
import '../pin_config/pin_config_view.dart';

/// Pick the board you're using and the transport you'll talk to it over.
class BoardSetupView extends StatelessWidget {
  const BoardSetupView({super.key});

  @override
  Widget build(BuildContext context) {
    final appState = context.watch<AppState>();
    final c = AppTheme.colorOf(context);
    final board = appState.selectedBoard;

    return Scaffold(
      backgroundColor: c.background,
      appBar: AppBar(title: const Text('Setup')),
      body: SingleChildScrollView(
        padding: EdgeInsets.all(AppTheme.spacing.lg),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SectionHeader(
              title: 'Choose your board',
              subtitle: 'Pick the module you wired the battery to.',
            ),
            SizedBox(height: AppTheme.spacing.lg),
            for (final b in appState.boards) ...[
              _BoardCard(
                board: b,
                isSelected: board?.id == b.id,
                onTap: () => appState.selectBoard(b),
              ),
              SizedBox(height: AppTheme.spacing.lg),
            ],
            if (board != null) ...[
              _TransportSection(board: board),
              SizedBox(height: AppTheme.spacing.lg + AppTheme.spacing.sm),
              PrimaryButton(
                onPressed: () => Navigator.of(context).push(
                  MaterialPageRoute(builder: (_) => const PinConfigView()),
                ),
                child: const Text('Configure'),
              ),
            ],
          ],
        ),
      ),
    );
  }
}

class _BoardCard extends StatelessWidget {
  final Board board;
  final bool isSelected;
  final VoidCallback onTap;

  const _BoardCard({
    required this.board,
    required this.isSelected,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    final c = AppTheme.colorOf(context);

    return GestureDetector(
      onTap: onTap,
      behavior: HitTestBehavior.opaque,
      child: AppCard(
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(board.name,
                      style: AppTheme.font.headline
                          .copyWith(color: c.textPrimary)),
                  SizedBox(height: AppTheme.spacing.xs),
                  Text(board.chip.displayName,
                      style: AppTheme.font.caption.copyWith(color: c.brand)),
                  SizedBox(height: AppTheme.spacing.xs),
                  Text(board.summary,
                      style: AppTheme.font.footnote
                          .copyWith(color: c.textSecondary)),
                  SizedBox(height: AppTheme.spacing.xs + AppTheme.spacing.xxs),
                  Wrap(
                    spacing: AppTheme.spacing.xs,
                    runSpacing: AppTheme.spacing.xs,
                    children: [
                      for (final t in board.supportedTransports)
                        TransportBadge(transport: t),
                    ],
                  ),
                ],
              ),
            ),
            SizedBox(width: AppTheme.spacing.md),
            Icon(
              isSelected ? Icons.check_circle : Icons.circle_outlined,
              color: isSelected ? c.brand : c.border,
              size: 28,
            ),
          ],
        ),
      ),
    );
  }
}

class _TransportSection extends StatelessWidget {
  final Board board;

  const _TransportSection({required this.board});

  @override
  Widget build(BuildContext context) {
    final appState = context.watch<AppState>();

    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        const SectionHeader(
          title: 'Transport',
          subtitle: 'How the app talks to this board.',
        ),
        SizedBox(height: AppTheme.spacing.sm),
        SegmentedPicker<FlashTransport>(
          options: board.supportedTransports,
          selection: board.supportedTransports.contains(appState.activeTransport)
              ? appState.activeTransport
              : board.supportedTransports.first,
          labelOf: (t) => t.displayName,
          onChanged: (t) => appState.activeTransport = t,
        ),
      ],
    );
  }
}
